### ISSUE TITLE
AutoClose doesn't work on update

### ISSUE THREAD

**Issue by jshap2000** [fkhadra/react-toastify#810]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

When updating a toast, the toast loses it's ability to autoclose (even when including autoclose in the update body). You must hover  directly over the toast to close it.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your CodeSandbox (https://codesandbox.io/s/new) example below:**

Create a toast. Then try to update that toast before it autocloses. The toast will now never autoclose unless you manually hover over it.

**What is the expected behavior?**

It should autoclose at a new clip equivalent to the time of update.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

Chrome. This has never worked and seems to correlate with other autoclose bugs.

**Comment by fkhadra** [comment]
Hey @jshap2000 could you reproduce the issue in codesandbox please, this would speedup the fix if needed :)

**Comment by jshap2000** [comment]
https://codesandbox.io/s/dreamy-ganguly-hy9z5p?file=/src/App.js

**Comment by jshap2000** [comment]
https://user-images.githubusercontent.com/46686400/182461988-8dd78bdd-1a1b-4816-b6c9-6751761b9c8e.mov

Hope this video helps also. autoclose pauses until you hover over it.

**Comment by fkhadra** [comment]
@jshap2000 it seems that the codesanbox you shared has not been saved :D

**Comment by jshap2000** [comment]
https://codesandbox.io/s/react-toastify-bug-replication-hy9z5p

**Comment by jshap2000** [comment]
Let me know if that link works, try replicating the behavior in the video.

**Comment by fkhadra** [comment]
@jshap2000 it seems that the next version(v9.0.8 will be released at the end of the week) will fix this issue. [![Edit Fix #810](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/fix-810-jv3qwt?fontsize=14&hidenavigation=1&theme=dark)

May I ask you to test in your project as well?

```sh
yarn add react-toastify@9.0.7-fix-autoclose-2
// or with npm
npm i react-toastify@9.0.7-fix-autoclose-2
```

**Comment by jshap2000** [comment]
Yup, the fix looks good!

**Comment by fkhadra** [comment]
@jshap2000 thanks for testing! Expect a new release by the end of the week

**Comment by jshap2000** [comment]
@fkhadra Sorry to circle back on this but there is still one slight bug in the update. On the very first toast it seems like there is a slight delay before autoclose progress begins. If you call update in that moment before autoclose begins the next call to update will hang the autoclose (it will never finish). 

Posted a video of first the bug and then it working as long as autoclose progress begins ...

https://user-images.githubusercontent.com/46686400/182883903-0b5c2955-2239-4020-b1cc-50eb0ee84a45.mov

**Comment by fkhadra** [comment]
@jshap2000 I'm having a hard time reproducing the issue. Am I doing something wrong?

https://user-images.githubusercontent.com/5574267/182886177-646ace5e-1d7a-4949-97fa-97c6a3bf356a.mov

**Comment by jshap2000** [comment]
Actually it might just have been the animation time length I was using.

**Comment by fkhadra** [comment]
The progress bar start only when the entrance animation is done

**Comment by jshap2000** [comment]
It seems like if the progress bar hasn't started yet, on a call to update, the progress bar will never start.

**Comment by jshap2000** [comment]
Is it possible to make it so that even if the progress bar hasn't started, on a call to update, it will still be able to start without having to manually hover over.

**Comment by fkhadra** [comment]
Using https://codesandbox.io/s/react-toastify-bug-replication-forked-z4lkdf I can confirm that the issue is coming from the animation timing.
```css
.fade-in {
  animation: fade-in 5.2s cubic-bezier(0.39, 0.575, 0.565, 1) both;
}
```
Just to give a bit of context. We don't want the autoclose timer to start before the entrance animation is complete, 
because if the autoClose delay is shorter than the entrance animation, you would never see the notification. 
That's why we wait for the entrance animation to complete before starting the autoClose timer.

For your case, I suggest tweaking the animation duration. 
Regarding
> Is it possible to make it so that even if the progress bar hasn't started, on a call to update, it will still be able to start without having to manually hover over.

I'll see what is possible to do without introducing a new API. If it's too much work I'll probably postpone to next minor release

**Comment by jshap2000** [comment]
Ok thanks again for your help with all this.

**Comment by fkhadra** [comment]
You're welcome
